gridstatus/miso_api.py

```diff
@@ -1142,6 +1143,203 @@ def get_real_time_generation_fuel_type_hourly(
+                days=1
+            )
+
+        date_str = date.strftime("%Y-%m-%d")
+
+        url = f"{BASE_LOAD_GENERATION_AND_INTERCHANGE_URL}/real-time/{date_str}/demand/actual?timeResolution={time_resolution}"
+
+        data_list = self._get_url(
+            url,
+            product=LOAD_GENERATION_AND_INTERCHANGE_PRODUCT,
+            verbose=verbose,
+        )
+
+        df = self._data_list_to_df(
+            data_list,
+        )
+
+        if "interval" in df.columns:
+            df = df.drop(columns=["interval"])
+
+        df["region"] = df["region"].str.upper()
+
+        df = df.rename(
+            columns={"region": "Region", "load": "Load"},
+        )
+
+        data = df.reset_index()
+
+        data = data[data["Interval Start"] >= date]
+
+        if end is not None:
+            data = data[data["Interval End"] <= end]
+
+        for col in data.columns:
+            if col not in ["Interval Start", "Interval End", "Region"]:
+                data[col] = data[col].astype(float)
+
+        return data[["Interval Start", "Interval End", "Region", "Load"]].reset_index(
+            drop=True
+        )
+
+    @support_date_range(frequency="DAY_START")
+    def get_actual_load_hourly(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        return self._get_actual_load(
+            date,
+            end=end,
+            verbose=verbose,
+            time_resolution=HOURLY_RESOLUTION,
+        )
+
+    @support_date_range(frequency="DAY_START")
+    def get_actual_load_daily(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+    ) -> pd.DataFrame:
+        return self._get_actual_load(
+            date,
+            end=end,
+            verbose=verbose,
+            time_resolution=DAILY_RESOLUTION,
+        )
+
+    def _get_medium_term_load_forecast(
+        self,
+        date: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp],
+        end: str | pd.Timestamp | tuple[pd.Timestamp, pd.Timestamp] | None = None,
+        verbose: bool = False,
+        init: str | pd.Timestamp | None = None,
+        time_resolution: str = HOURLY_RESOLUTION,
+    ) -> pd.DataFrame:
+        """
+        init: Optional init date to get forecast from.
+            It must be earlier than both:
+                - the current date (now)
+                - and the forecast date (the date for which you're requesting predictions).
+
+            If you don't specify this init date:
+                - When your requested date is before today, it defaults to date - 1 day
+                - When your requested date is today or in the future, it defaults to today - 1 day
```

Comment:
We generally try to provide a "Publish Time" column with our forecast datasets. I think we can use this logic to provide at least an approximate publish time even though MISO doesn't provide it in the API. 

```python
miso_publish_time = min(
            date.normalize() - pd.DateOffset(days=1),
            pd.Timestamp.now(tz=self.default_timezone).normalize()
            - pd.DateOffset(days=1),
        )
df["Publish Time"] = init.normalize() if init is not None else miso_publish_time
```

(using the start of the day since we don't have a specific time).